tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/api/validation/PersonValidator.java

```diff
@@ -238,14 +238,17 @@ public void validateForBulk(List<PersonDTO> personDtos) {
           .map(String::trim)
           .collect(Collectors.toList());
 
-      personDto.setRole(String.join(",", roles));
+      Map<String, String> rolesMatch = referenceService.rolesMatch(roles, true);
+      List<String> matchedRoles = rolesMatch.entrySet().stream()
+          .map(entry -> entry.getValue().isEmpty() ? entry.getKey() : entry.getValue())
```

Comment:
Looks like there's a potential for an NPE here.  You could use `StringUtils.isEmpty()`.

Reply:
I _could_ guard it against NPEs to be extra careful. A null value can only exist if it's a case-insensitive match to a null key, which can't exist in a TreeMap, so it'd break in Reference first, either when trying to build the TreeMap, or when looking for matches for null strings. Also it would imply that the Role cell in the excel file has been left null, which would make the whole file be rejected before processing it at all.
But guarding it against NPEs is good anyway, in case the software around it changes in the future, so I've changed these lines to use StringUtils.